src/com/oltpbenchmark/api/Worker.java

```diff
@@ -282,14 +329,28 @@ public final void run() {
                     // Do nothing
             }
 
+
             // PART 3: Execute work
 
             // TODO: Measuring latency when not rate limited is ... a little
             // weird because if you add more simultaneous clients, you will
             // increase latency (queue delay) but we do this anyway since it is
             // useful sometimes
 
-            long start = pieceOfWork.getStartTime();
+            if (this.wrkld.getUseKeyingTime()) {
+                // Wait for the keying time which is a fixed number for every type of transaction.
+                long keying_time_msecs = getKeyingTime(transactionTypes.getType(pieceOfWork.getType()));
+                try {
+                    long start_sleep = System.nanoTime();
```

Comment:
Do we need this variable and the following LOG.info to measure the sleep time? We should get rid of these I feel.

Reply:
I did this for debugging.
WOuldn't be part of the final code.